### PR TITLE
appraiser-rounds-and-links

### PR DESCRIPTION
**Pull request by Space-tourist**
rounds count and single url added to story

### CODE REVIEW

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/AddStoryCommandHandler.cs [hunk 1]
```diff
@@ -28,6 +29,20 @@ public AddStoryCommandHandler(
     public async Task<CommandResult> Handle(AddStoryCommand command, CancellationToken token)
     {
         ArgumentNullException.ThrowIfNull(command);
+        
+        var separator = ' ';
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/AddStoryCommandHandler.cs:33]: const char

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Services/AddStoryCommandCreator.cs [hunk 2]
```diff
@@ -24,27 +24,14 @@ public async Task<IDialogCommand> Create(
...
         ArgumentNullException.ThrowIfNull(teamContext);
-
-        var separator = ' ';
+        
         var teammates = await _teamAccessor.GetTeammates(teamContext.TeamId, DateTimeOffset.UtcNow, token);
-        var storyItems = messageContext.Text.Split(separator);
-        var links = new List<string>();
-        var text = new List<string>();
-
-        foreach (var storyItem in storyItems)
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Services/AddStoryCommandCreator.cs:34]: Why the logic has been moved to command handler?

File: src/Inc.TeamAssistant.Appraiser.Application/Services/SummaryByStoryBuilder.cs [hunk 3]
```diff
@@ -161,4 +164,15 @@ private string BuildLinkForDashboard(Guid teamId, LanguageId languageId)
             languageId.Value,
             teamId.ToString("N"));
     }
+
+    private async Task AddRoundsInfo(StringBuilder builder, SummaryByStory summary)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(summary);
+        
+        builder.AppendLine();
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/Services/SummaryByStoryBuilder.cs:173]: Please add variable for await _messageBuilder.Build(Messages.Appraiser_NumberOfRounds, summary.LanguageId) and write single line: builder.AppendLine($"{newVariable} {summary.RoundsCount}")

File: src/Inc.TeamAssistant.Appraiser.Domain/Story.cs [hunk 4]
```diff
@@ -76,9 +76,7 @@ public void AddStoryForEstimate(StoryForEstimate storyForEstimate)
 
 	public void AddLink(string link)
 	{
-		ArgumentException.ThrowIfNullOrWhiteSpace(link);
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Domain/Story.cs:79]: Please check variable link

File: src/Inc.TeamAssistant.Gateway/Services/Integrations/EstimatesService.cs [hunk 5]
```diff
@@ -41,15 +41,14 @@ public async Task StartEstimate(StartEstimateRequest request, CancellationToken
...
+        var url = string.IsNullOrWhiteSpace(request.IssueUrl)
+            ? string.Empty
+            : request.IssueUrl; 
         var command = new AddStoryCommand(
             messageContext,
             context.TeamId,
             context.TeamProperties.GetStoryType(),
-            title,
-            links,
+            $"{title} {url}",
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Gateway/Services/Integrations/EstimatesService.cs:51]: Please use two field in the command for title and url

File: src/Inc.TeamAssistant.Migrations/2024_11_25_0_AddUrl.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+public sealed class AddUrl  : Migration
+{
+    public override void Up()
+    {
+        Create
+            .Column("url")
+            .OnTable("stories")
+            .InSchema("appraiser")
+            .AsString()
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Migrations/2024_11_25_0_AddUrl.cs:15]: Please set length 2000 and add validation message

File: src/Inc.TeamAssistant.Stories/Features/AssessmentSession/AssessmentSession.stories.razor [hunk 7]
```diff
@@ -103,7 +102,9 @@
             EstimateEnded: true,
             Mean: "9.6",
             Median: "8",
-            AcceptedValue: string.Empty));
+            AcceptedValue: string.Empty,
+            RoundsCount: 1,
+            Url: "https://amazon.com"));
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Stories/Features/AssessmentSession/AssessmentSession.stories.razor:107]: Where is my favorite company - OZON? =)

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/AddStoryCommandHandler.cs (hunk 1 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/AddStoryCommandHandler.cs]: changes reverted

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Services/AddStoryCommandCreator.cs (hunk 2 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Services/AddStoryCommandCreator.cs]: fixed

File: src/Inc.TeamAssistant.Appraiser.Application/Services/SummaryByStoryBuilder.cs (hunk 3 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/Services/SummaryByStoryBuilder.cs:173]: fixed

File: src/Inc.TeamAssistant.Appraiser.Domain/Story.cs (hunk 4 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Domain/Story.cs:79]: fixed

File: src/Inc.TeamAssistant.Gateway/Services/Integrations/EstimatesService.cs (hunk 5 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Gateway/Services/Integrations/EstimatesService.cs]: fixed

File: src/Inc.TeamAssistant.Migrations/2024_11_25_0_AddUrl.cs (hunk 6 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Migrations/2024_11_25_0_AddUrl.cs]: fixed

File: src/Inc.TeamAssistant.Stories/Features/AssessmentSession/AssessmentSession.stories.razor (hunk 7 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Stories/Features/AssessmentSession/AssessmentSession.stories.razor]: fixed:)

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs [hunk 8]
```diff
@@ -18,10 +22,35 @@ public AddStoryCommandValidator()
...
         RuleFor(e => e.Teammates)
             .NotEmpty();
     }
+
+    private async Task CheckLinks(
+        IReadOnlyCollection<string> links,
+        ValidationContext<AddStoryCommand> context,
+        CancellationToken token)
+    {
+        var errorMessage = string.Empty;
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs:37]: Please don't reuse variables. Remove errorMessage and add multiple use context.AddFailure

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs [hunk 9]
```diff
@@ -18,10 +22,35 @@ public AddStoryCommandValidator()
...
+        if (links.Count == 1)
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs:38]: What happens if you do not specify a reference to a story?

File: src/Inc.TeamAssistant.Appraiser.DataAccess/StoryRepository.cs [hunk 10]
```diff
@@ -75,8 +77,10 @@ ON CONFLICT (id) DO UPDATE SET
                 language_id = story.LanguageId.Value,
                 title = story.Title,
                 external_id = story.ExternalId,
-                links = JsonSerializer.Serialize(story.Links),
-                total_value = story.TotalValue
+                links = JsonSerializer.Serialize(Array.Empty<string>()),
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.DataAccess/StoryRepository.cs:80]: Why don't we delete the field links?

File: src/Inc.TeamAssistant.Migrations/2024_11_25_0_AddUrl.cs [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            .AsString(2000)
+            .Nullable();
+        
+        Execute.Sql(
+            """
+            UPDATE appraiser.stories
+            SET url = links->>0;
+            """,
+            "Add single url to story");
+    }
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Migrations/2024_11_25_0_AddUrl.cs:24]: Add remove field links

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs [hunk 12]
```diff
@@ -18,10 +22,35 @@ public AddStoryCommandValidator()
...
-            .NotEmpty();
+        RuleFor(e => e.Links)
+            .CustomAsync(CheckLinks);
         
         RuleFor(e => e.Teammates)
             .NotEmpty();
     }
+
+    private async Task CheckLinks(
+        IReadOnlyCollection<string> links,
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs:33]: Add check arguments

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs (hunk 8 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs]: fixed

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs (hunk 9 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs]: fixed

File: src/Inc.TeamAssistant.Appraiser.DataAccess/StoryRepository.cs (hunk 10 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.DataAccess/StoryRepository.cs]: fixed

File: src/Inc.TeamAssistant.Migrations/2024_11_25_0_AddUrl.cs (hunk 11 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Migrations/2024_11_25_0_AddUrl.cs:29]: fixed

File: src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs (hunk 12 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/CommandHandlers/AddStory/Validators/AddStoryCommandValidator.cs:33]: fixed

**Review by dyatlov-a — APPROVED** [review]

**Pull request merged**